spring-pulsar-docs/src/main/asciidoc/pulsar.adoc

```diff
@@ -1222,6 +1262,116 @@ PulsarTopic partitionedTopic {
 ----
 ====
 
+[[pulsar-headers]]
+==== Pulsar Headers
+
+In this section, we will see how we can use the message metadata that Pulsar provides in its Message as Pulsar headers in a Spring application.
+First, let's examine all the available Pulsar headers from the message metadata.
+
+.[.underline]#Aailable **Pulsar Message Metadata as Spring Headers**#
+[%collapsible]
+====
+https://github.com/spring-projects-experimental/spring-pulsar/blob/main/spring-pulsar/src/main/java/org/springframework/pulsar/support/PulsarHeaders.java[Click here] to see the available Pulsar headers.
+====
+
+===== Accessing Pulsar Headers in Single Record based Consumer
+
+Here is an example of how you can access the various Pulsar Headers in an application that is using single record mode of consuming.
+
+====
+[source,java]
+----
+@PulsarListener(topics = "simpleListenerWithHeaders")
+void simpleListenerWithHeaders(String data, @Header(PulsarHeaders.MESSAGE_ID) MessageId messageId,
+                @Header(PulsarHeaders.RAW_DATA) byte[] rawData,
+                @Header("foo") String foo) {
+
+}
+----
+====
+
+In this example, we are consuming the data as `String`.
```

Comment:
It may be helpful to

1. say "In the above example..." as it may not be clear if it is talking about the next example. 
2. the fact that the data is string is probably not pertinent to the header infos
3. I think we can simplify this a bit. 

Maybe something like: 
```java
In the above example we are accessing the values for the `messageId` and `rawData` message metadata as well as a custom message property named "foo". The  Spring `@Header` annotation is used for each header field.
```

Wdyt?